### PR TITLE
chore: remove knative support

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Removes knative support.

**Which issue this PR fixes**:

Related #2813

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -81,6 +81,8 @@ Adding a new version? You'll need three changes:
 
 - Removed feature gate `CombinedServices`. The feature is enabled and it can't be changed.
   [#4743](https://github.com/Kong/kubernetes-ingress-controller/pull/4743)
+- Removed knative support.
```
Comment (programmer04) [inline CHANGELOG.md:84]:
```suggestion
- Removed Knative support.
```
[It's capitalized](https://github.com/knative/docs?tab=readme-ov-file#website)

**Review by programmer04 — APPROVED** [review]
🧹 
Also, Helm chart has to be adjusted to not install Knative-related stuff for KIC `>=3.0`

File: .github/workflows/_integration_tests.yaml [hunk 2]
```diff
@@ -51,11 +51,7 @@ jobs:
             test: postgres
           - name: feature-gates
             test: dbless
-            feature_gates: "GatewayAlpha=true"
-          - name: dbless-knative
-            test: dbless.knative
-          - name: postgres-knative
-            test: postgres.knative
+            feature_gates: "GatewayAlpha=true,CombinedRoutes=false"
```
Comment (programmer04) [inline .github/workflows/_integration_tests.yaml:54]:
```suggestion
            feature_gates: "GatewayAlpha=true"
```
we don't have it anymore
- https://github.com/Kong/kubernetes-ingress-controller/pull/4749

**Review by programmer04 — CHANGES_REQUESTED** [review]

File: .github/workflows/_integration_tests.yaml (hunk 2 above)
Comment (pmalek) [reply .github/workflows/_integration_tests.yaml:54]: Alright then this becomes the same as `dbless-gateway-alpha` which means we can get rid of this

**Review by programmer04 — APPROVED** [review]
🕺 removing code is the best thing

**Pull request merged**